vllm/sequence.py

```diff
@@ -206,6 +229,10 @@ def hash_of_block(self, logical_idx: int) -> int:
     def num_hashed_tokens_of_block(self, logical_idx: int):
         return logical_idx * self.block_size + self.block_size
 
+    def on_recompute(self):
+        """Reset the sequence states for recomputation."""
+        self.data.reset_num_computed_tokens()
```

Comment:
maybe call this `reset_state` so it's easier to understand. `recompute` is when reset is needed (currently the only scenario but very specific) but this operation is just resetting.

Reply:
The confusing part (and the reason why I chose more specific term here) is that it actually doesn't "reset" the state because reset sounds like output_token_ids are also cleared. But this API doesn't do that. What about reset_state_for_recompute?